base/applications/cmdutils/more/more.c

```diff
@@ -51,6 +52,144 @@ HANDLE hKeyboard;
 /* Enable/Disable extensions */
 BOOL bEnableExtensions = TRUE; // FIXME: By default, it should be FALSE.
 
+#define FLAG_HELP (1 << 0)
+#define FLAG_E (1 << 1)
+#define FLAG_C (1 << 2)
+#define FLAG_P (1 << 3)
+#define FLAG_S (1 << 4)
+#define FLAG_Tn (1 << 5)
+#define FLAG_PLUSn (1 << 6)
+
+static DWORD s_dwFlags = 0;
+static DWORD s_nTabWidth = 8;
+static DWORD s_nNextLineNo = 0;
+static BOOL s_bPrevLineIsBlank = FALSE;
+static UINT s_nPromptID = IDS_CONTINUE_PROGRESS;
+static BOOL s_bDoNextFile = FALSE;
+
+static BOOL IsBlankLine(IN PCWCH line, IN DWORD cch)
+{
+    DWORD ich;
+    WORD wType;
+    for (ich = 0; ich < cch; ++ich)
+    {
+        wType = 0;
+        GetStringTypeW(CT_CTYPE1, &line[ich], 1, &wType);
+        if (!(wType & (C1_BLANK | C1_SPACE)))
+            return FALSE;
+    }
+    return TRUE;
+}
+
+static BOOL __stdcall
+MorePagerExpandTab(
+    IN OUT PCON_PAGER Pager,
+    IN PCWCH line,
+    IN DWORD cch,
+    IN OUT PDWORD pdwFlags)
+{
+    PWCHAR psz;
+    DWORD ich1, ich2, cch2, spaces;
+    BOOL ret;
+    DWORD iColumn = Pager->iColumn;
+
+    /* Calculate expanded length */
+    for (ich1 = ich2 = 0; ich1 < cch; ++ich1)
+    {
+        if (line[ich1] == L'\t')
+        {
+            spaces = s_nTabWidth - (iColumn % s_nTabWidth);
+            iColumn += spaces;
+            ich2 += spaces;
+        }
+        else if (line[ich1] == L'\n')
+        {
+            iColumn = 0;
+            ++ich2;
+        }
+        else
+        {
+            ++iColumn;
+            ++ich2;
+        }
+    }
+
+    cch2 = ich2;
+    psz = malloc((cch2 + 1) * sizeof(WCHAR));
+    if (!psz)
+        return FALSE;
+
+    /* Store to buffer */
+    iColumn = Pager->iColumn;
+    // NOTE: Also, it would have been: ich2 <= cch2
+    for (ich1 = ich2 = 0; ich1 < cch /*&& ich2 < cch2*/; ++ich1)
+    {
+        if (line[ich1] == L'\t')
+        {
+            spaces = s_nTabWidth - (iColumn % s_nTabWidth);
+            iColumn += spaces;
+            while (spaces-- > 0)
+            {
+                psz[ich2++] = L' ';
+            }
+        }
+        else if (line[ich1] == L'\n')
+        {
+            iColumn = 0;
+            psz[ich2++] = line[ich1];
+        }
+        else
+        {
+            ++iColumn;
+            psz[ich2++] = line[ich1];
```

Comment:
Note: For some reason, code analyzer says here:
```
Warning C6386: Buffer overrun while writing to 'psz':  the writable size is '((cch2+1))*sizeof(WCHAR)' bytes, but '4' bytes might be written.
```
My confidence in this is quite low, since this for-loop appears to exactly do the same thing as the one above, and the psz buffer should be correctly sized.